### PR TITLE
Ciam 2165 make generic metadata

### PR DESCRIPTION
**Pull request by stelewis-redhat**

### CODE REVIEW

File: server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java [hunk 1]
```diff
@@ -620,6 +620,28 @@ public static CredentialMetadataRepresentation toRepresentation(CredentialMetada
         return rep;
     }
 
+    public static CredentialMetadataRepresentation toRepresentation(CredentialMetadata credentialMetadata) {
```
Comment (stelewis-redhat) [inline server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java]: Not sure why this was redone. Was already in parent branch.

File: services/src/main/java/org/keycloak/services/resources/account/AccountCredentialResource.java [hunk 2]
```diff
@@ -139,7 +139,7 @@ public boolean isRemoveable() {
             return removeable;
         }
 
-        public List<CredentialMetadataRepresentation> getUserCredentials() {
+        public List<CredentialMetadataRepresentation> getUserCredentialMetadatas() {
```
Comment (stelewis-redhat) [inline services/src/main/java/org/keycloak/services/resources/account/AccountCredentialResource.java:142]: This is the important fix. Jackson serialization/deserialization couldn't find the getter for the member.

File: server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java (hunk 1 above)
Comment (zwitter1) [reply server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java]: ya i remember reading it

**Review by zwitter1 — APPROVED** [review]
I think this will work!

**Pull request merged**